frontendGraphImpl/src/main/java/com/virtuslab/gitmachete/frontend/graph/impl/repository/RepositoryGraph.java

```diff
@@ -1,100 +1,78 @@
-package com.virtuslab.gitmachete.frontend.graph.repository;
+package com.virtuslab.gitmachete.frontend.graph.impl.repository;
 
 import java.util.Collection;
-import java.util.Collections;
 
 import com.intellij.util.SmartList;
-import com.intellij.vcs.log.graph.PrintElement;
-import com.intellij.vcs.log.graph.api.EdgeFilter;
-import com.intellij.vcs.log.graph.api.elements.GraphEdge;
-import com.intellij.vcs.log.graph.api.elements.GraphEdgeType;
 import io.vavr.Tuple;
 import io.vavr.Tuple2;
 import io.vavr.collection.List;
-import lombok.Getter;
 import org.checkerframework.checker.index.qual.NonNegative;
 
-import com.virtuslab.gitmachete.backend.api.NullRepository;
-import com.virtuslab.gitmachete.frontend.graph.GraphElementManager;
-import com.virtuslab.gitmachete.frontend.graph.nodes.IGraphNode;
-import com.virtuslab.gitmachete.frontend.graph.print.PrintElementGeneratorImpl;
+import com.virtuslab.gitmachete.frontend.graph.api.elements.GraphEdge;
+import com.virtuslab.gitmachete.frontend.graph.api.items.IGraphItem;
+import com.virtuslab.gitmachete.frontend.graph.api.print.elements.IPrintElement;
+import com.virtuslab.gitmachete.frontend.graph.api.repository.IRepositoryGraph;
+import com.virtuslab.gitmachete.frontend.graph.impl.print.PrintElementGenerator;
 
-public class RepositoryGraph {
-  @Getter
-  @SuppressWarnings("ConstantName")
-  private static final RepositoryGraph nullRepositoryGraph = new RepositoryGraphBuilder()
-      .repository(NullRepository.getInstance()).build();
+public class RepositoryGraph implements IRepositoryGraph {
 
-  private final List<IGraphNode> nodes;
+  private final List<IGraphItem> items;
   private final List<List<Integer>> positionsOfVisibleEdges;
-
-  private final PrintElementGeneratorImpl printElementGenerator;
+  private final PrintElementGenerator printElementGenerator;
 
   @SuppressWarnings({"nullness:argument.type.incompatible", "nullness:assignment.type.incompatible"})
```

Comment:
Btw see https://github.com/VirtusLab/git-machete-intellij-plugin/pull/228 for how to get rid of that (you can apply it on your PR since it's a small change + the affected classes might change on your PR anyway)

Reply:
great, thanks, applied